PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/ValidationUtils.java

```diff
@@ -44,4 +51,22 @@ public static boolean isValidFunctionName(String name) {
     public static boolean isValidVersion(String version) {
         return version != null && version.matches(VERSION_REGEX);
     }
+
+    public static void validFunctionAppName(String subscriptionId, String functionAppName) {
+        if (StringUtils.isEmpty(subscriptionId)) {
+            throw new IllegalArgumentException("Subscription can not be null");
+        }
+        if (!isValidFunctionName(functionAppName)) {
+            throw new IllegalArgumentException("Function App names only allow alphanumeric characters, periods, underscores, hyphens and parenthesis and cannot end in a period.");
+        }
+        try {
+            final Azure azure = AuthMethodManager.getInstance().getAzureManager().getAzure(subscriptionId);
+            final FunctionApp target = azure.appServices().functionApps().getByResourceGroup(subscriptionId, functionAppName);
```

Comment:
The 1st parameter of getByResourceGroup should be  resourceGroupName?

Reply:
Is there any lightweight api rather than loading all function  app details to verify a function name existence?

Reply:
Changed to use `WebSiteManagementClientImpl.checkNameAvailability` to do the validation job